src/librustdoc/html/static/main.js

```diff
@@ -54,7 +54,8 @@
     }
 
     function browserSupportsHistoryApi() {
-        return window.history && typeof window.history.pushState === "function";
+        return document.origin != "null" &&
```

Comment:
Interesting, and I thought this was the standard. I will update & test in Firefox as well.